src/main/scala/Divider.scala

```diff
@@ -22,33 +22,33 @@ object Divider:
 
     val indexedBoards: List[(Board, Int)] = boards.zipWithIndex
 
-    val midGame = indexedBoards.collectFirst:
-      case (board, index)
-          if (majorsAndMinors(board) <= 10 ||
-            backrankSparse(board) ||
-            mixedness(board) > 150) =>
-        index
+    val midGame = indexedBoards.foldLeft(none[Int]):
+      case (None, (board, index)) =>
+        (majorsAndMinors(board) <= 10 ||
+          backrankSparse(board) ||
+          mixedness(board) > 150) option index
+      case (found, _) => found
```

Comment:
@lenguyenthanh is this change actually affects benches? If so, do you know why the manual fold is faster then `collectFirst`?

Reply:
interesting, I have no recollection about this change 😅, divider benchmark did improve, so probably yes, but I'm not sure why. And this could be a mistake as well. In order to solve this, I'll need to re-benchmark and check the bytecode output.

Reply:
did run the benchmark again, there is no different between foldLeft and collectFirst.